---
hide:
  - footer
---

The following software packaged were used for processing and analysis:

|Software|Citation|
|---|----|
|Bowtie2| Langmead B, et al., Nat Methods, 2012, DOI: 10.1038/nmeth.1923|
|Canvas| Chovanec P, et al., Nat Communications, 2021, DOI: 10.1038/s41467-021-22201-4|
|Chicago| Cairns J, et al., Genome Biology, 2016, DOI: 10.1186/s13059-016-0992-2|
|FastQC|https://www.bioinformatics.babraham.ac.uk/projects/fastqc/|
|FastQScreen|Wingett SW, et al., F1000Res, 2018, DOI: 10.12688/f1000research.15931.2|
|Gephi|Bastian M., Heymann S., Jacomy M. (2009). Gephi: an open source software for exploring and manipulating networks. International AAAI Conference on Weblogs and Social Media.|
|HICUP|Wingett SW, et al., F1000Res, DOI:10.12688/f1000research.7334.1|
|Hisat2|Kim D, et al., Nat Biotechnology, 2019, DOI: 10.1038/s41587-019-0201-4|
|Igraph|Csardi G, Nepusz T (2006). “The igraph software package for complex network research.” InterJournal, Complex Systems, 1695. https://igraph.org.|
|MultiQC|Ewels P, et al., Bioinformatics, 2016, DOI: 10.1093/bioinformatics/btw354|
|R|R Core Team (2021). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria.URL https://www.R-project.org/.|
|Seqmonk|https://www.bioinformatics.babraham.ac.uk/projects/seqmonk/|
|Tidyverse|Wickham H, et al., Journal of Open Source Software, DOI: 10.21105/joss.01686|
|Trim Galore|https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/, latest release DOI: 10.5281/zenodo.5127899|
|Uwot|McInnes L, et al., arXiv, 2018, DOI: 10.48550/arXiv.1802.03426|
|Weka|Eibe Frank, Mark A. Hall, and Ian H. Witten (2016). The WEKA Workbench. Online Appendix for "Data Mining: Practical Machine Learning Tools and Techniques", Morgan Kaufmann, Fourth Edition, 2016.|
